File: 2025/Company/Company.Web/Modules/ServerTypes/Company/CurrencyRow.ts
import { fieldsProxy } from '@serenity-is/corelib';

export interface CurrencyRow {
    CurrencyId?: number;
    CurrencyCode?: string;
    Name?: string;
    IsActive?: number;
    TenantId?: number;
}

export abstract class CurrencyRow {
    static readonly idProperty = 'CurrencyId';
    static readonly nameProperty = 'CurrencyCode';
    static readonly localTextPrefix = 'Company.Currency';

    static readonly deletePermission = 'PermissionKeys.Currency';
    static readonly insertPermission = 'PermissionKeys.Currency';
    static readonly readPermission = 'PermissionKeys.Currency';
    static readonly updatePermission = 'PermissionKeys.Currency';

    static readonly Fields = fieldsProxy<CurrencyRow>();
}

File: 2025/Company/Company.Web/Modules/Company/Currency/CurrencyRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace Company.Company;

[ConnectionKey("Company"), Module("Company"), TableName("[sales].[Currency]")]
[DisplayName("Currency"), InstanceName("Currency")]
[ReadPermission("PermissionKeys.Currency")]
[ModifyPermission("PermissionKeys.Currency")]
[ServiceLookupPermission("PermissionKeys.Currency")]
public sealed class CurrencyRow : Administration.LoggingRow<CurrencyRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Currency Id"), Identity, IdProperty]
    public int? CurrencyId { get => fields.CurrencyId[this]; set => fields.CurrencyId[this] = value; }

    [DisplayName("Currency Code"), Size(3), NotNull, QuickSearch, NameProperty]
    public string CurrencyCode { get => fields.CurrencyCode[this]; set => fields.CurrencyCode[this] = value; }

    [DisplayName("Name"), Size(50), NotNull]
    public string Name { get => fields.Name[this]; set => fields.Name[this] = value; }

    [DisplayName("Is Active"), NotNull]
    public short? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    [DisplayName("Tenant Id"), NotNull]
    public int? TenantId { get => fields.TenantId[this]; set => fields.TenantId[this] = value; }

    public class RowFields : Administration.LoggingRowFields
    {
        public Int32Field CurrencyId;
        public StringField CurrencyCode;
        public StringField Name;
        public Int16Field IsActive;
        public Int32Field TenantId;

    }
}